pkg/apis/kubermatic/v1/datacenter.go

```diff
@@ -680,6 +680,44 @@ type DatacenterSpecKubevirt struct {
 	// DNSConfig represents the DNS parameters of a pod. Parameters specified here will be merged to the generated DNS
 	// configuration based on DNSPolicy.
 	DNSConfig *corev1.PodDNSConfig `json:"dnsConfig,omitempty"`
+
+	// Images represents standard VM Image source.
+	Images *Images `json:"images,omitempty"`
+}
+
+var (
+	SupportedKubeVirtOS = map[providerconfig.OperatingSystem]*struct{}{
```

Comment:
Having supportedKubeVirtOS as map instead of array doesn't give you any performance improvement in that case. This is too small. Map access is O(1) but you still have to compute the hash, search for the right bucket and element, it doesn't mean that it's more efficient than having just array and compare elements.

Can you move array of supported os directly to validation file?

Reply:
The map is small but checking if an `os` exists in Map is easy as compared to an array where you iterate through it, looks ugly, to be honest, plus extra lines of code.

Also, this Map is used in API (Dashboard) to return only valid supported OS.

Reply:
I guess that was copied from MC :) Once I have some time free i will start refactoring there.

Reply:
> looks ugly, to be honest
I disagree, but your call. Unfortunately Go doesn't implement set data struct and as I can read it's acceptable form to use map as you did (for set replacement).

However, I don't think it has to be a global variable, can you move it to validation function directly?

Reply:
I think you missed the point I mentioned earlier :)
As on the dashboard repo, we have kkp dependency, we re-use this Map from `datacenter.go` for list image API also, that is why I made it global.

Reply:
Ah yes, then I missed.

Reply:
In that case could we move this var to kubevirt provider code?

Reply:
As some kubevirt provider code is duplicated in kkp & dashboard both, if I move this var to provider then it has to be duplicated across both the repos.

Reply:
understood.